File: frontend/src/assets/icons/min.py
#!/usr/bin/env python3
import os
import glob
import subprocess

if __name__ == "__main__":
    svg_files = []
    for file in glob.glob("svg_raw/*.svg"):
        svg_files.append(file)
    for f in svg_files:
        command_str = "svgo {} -o svg_min/{}".format(f,
                os.path.basename(os.path.splitext(f)[0])+".min.svg")
        print(command_str)
        process = subprocess.Popen(command_str.split(), stdout=subprocess.PIPE)
        output, error = process.communicate()

File: frontend/src/assets/icons/min.js
const fs = require('fs');
const path = require('path');
const process = require('process');
const { exec } = require('child_process');

const svgFolder = 'svg_raw';
const svgMinFolder = 'svg_min';

fs.readdir(svgFolder, (err, files) => {
  if (err) {
    console.error(err);
    process.exit(1);
  }
  if (!fs.existsSync(svgMinFolder)) {
    console.error(`svg output folder not exist: ${svgMinFolder}`);
    process.exit(1);
  }
  files.forEach((file) => {
    if (path.extname(file) == '.svg') {
      const command_str = `svgo ${svgFolder}/${file} -o ${svgMinFolder}/${path.basename(
        file,
        '.svg'
      )}.min.svg`;
      console.log(command_str);
      exec(command_str, (err, stdout, stderr) => {
        if (err) {
          console.error(err);
        } else {
          if (stdout) {
            console.log(`stdout: ${stdout}`);
          }
          //console.log(`stderr: ${stderr}`);
        }
      });
    }
  });
});
